services/cd-service/pkg/service/git.go

```diff
@@ -153,10 +157,95 @@ func (s *GitServer) GetCommitInfo(ctx context.Context, in *api.GetCommitInfoRequ
 	for _, dir := range dirs {
 		touchedApps = append(touchedApps, dir.Name())
 	}
-	sort.Strings(touchedApps);
+	sort.Strings(touchedApps)
+
+	events, err := s.GetEvents(ctx, fs, in.CommitHash, commitPath)
+	if err != nil {
+		return nil, err
+	}
 
 	return &api.GetCommitInfoResponse{
 		CommitMessage: commitMessage,
-		TouchedApps: touchedApps,
+		TouchedApps:   touchedApps,
+		Events:        events,
 	}, nil
-}
\ No newline at end of file
+}
+
+func (s *GitServer) GetEvents(ctx context.Context, fs billy.Filesystem, hash string, commitPath string) ([]*api.Event, error) {
+	errorTemplate := func(message string, err error) error {
+		return fmt.Errorf("could not get events for hash '%s' in path '%s' - error %w", hash, commitPath, err)
```

Comment:
Maybe this is standard in the codebase, but it feels like this is an antipattern: You should not automatically wrap all errors that you return from a given function. The caller of the function should decide whether it should be wrapped to provide more context. Otherwise you'll end up with very long, redundant error messages in a lot of cases.

Reply:
> The caller of the function should decide whether it should be wrapped

I would argue that *I am* the caller here, wrapping errors from called functions.
If I have the choice between "sometimes too little information" and "sometimes too much information", I'll take the second one.
How would you do this?